lib/crypto/c_src/mac.c

```diff
@@ -114,10 +127,25 @@ void init_mac_types(ErlNifEnv* env)
 
     for (p = mac_types; p->name.str; p++) {
 	p->name.atom = enif_make_atom(env, p->name.str);
+#if defined(HAS_3_0_API)
+        p->evp_mac = EVP_MAC_fetch(NULL, p->fetch_name, NULL);
+#endif
     }
     p->name.atom = atom_false;  /* end marker */
 }
 
+void fini_mac_types(void)
+{
+#if defined(HAS_3_0_API)
+    struct mac_type_t* p = mac_types;
+
+    for (p = mac_types; p->name.str; p++) {
+        EVP_MAC_free(p->evp_mac);
```

Comment:
add p->evp_mac = NULL ?

Reply:
You can set to NULL for clarity if you want. But I don't think it will have any practical effect. When fini_mac_types() is called the NIF lib is doomed to be unloaded. No more NIF calls can be made.